src/Razor/src/Microsoft.VisualStudio.LanguageServer.ContainedLanguage/DefaultLSPDocumentFactory.cs

```diff
@@ -57,7 +57,11 @@ private IReadOnlyList<VirtualDocument> CreateVirtualDocuments(ITextBuffer hostDo
         {
             if (factory.Metadata.ContentTypes.Any(ct => hostDocumentBuffer.ContentType.IsOfType(ct)))
             {
-                if (factory.Value.TryCreateFor(hostDocumentBuffer, out var virtualDocument))
+                if (factory.Value.TryCreateMultipleFor(hostDocumentBuffer, out var newVirtualDocuments))
```

Comment:
Still reading, but when would `TryCreateMultipleFor` fail? Could it just create a single?

Reply:
I think I understand more now. We're using this as a gate for if multiple is supported. If so, it should be called instead of the single version?

Reply:
I'm not sure I follow. It is called instead of the single version.

The base implementation could call `TryCreateFor`, but I wanted to be super conservative about not changing existing code paths.